microservice-group/build.gradle

```diff
@@ -0,0 +1,146 @@
+    }
+}
+
+task url {
+    doLast { println groupURL }
+}
+
+test {
+    group 'Verification'
+    reports.html.destination = file("$buildDir/reports/unit")
+    reports.junitXml.destination = file("$buildDir/test-reports/unit")
+    exclude '**/test/**'
+
+    // run every time when called
+    outputs.upToDateWhen { false }
+}
+
+task integrationTest(type: Test) {
+    group 'Verification'
+    description 'Runs the integration tests.'
+    reports.html.destination = file("$buildDir/reports/it")
+    reports.junitXml.destination = file("$buildDir/test-reports/it")
+    include '**/test/**'
+
+    systemProperties = ['liberty.test.group.service.url': groupServiceURL,
+                        'mongo.test.hostname': groupMongoHostname,
+                        'mongo.test.port': groupMongoPort,
+                        'jwt.issuer': jwtIssuer,
+                        'javax.net.ssl.keyStorePassword': "secret"]
+
+    // run every time when called
+    outputs.upToDateWhen { false }
+}
+
+task copyKeystore (type: Copy) {
+    configurations.keystore.each {
+        from(zipTree(it)) {
+            include "keystore.jceks"
+        }
+    }
+    into "$buildDir/wlp/usr/servers/${wlpServerName}/resources/security"
+    outputs.file("$buildDir/wlp/usr/servers/${wlpServerName}/resources/security/keystore.jceks")
+}
+
+task testKeystore(type: Copy) {
+    dependsOn 'copyKeystore'
+    from "$buildDir/wlp/usr/servers/${wlpServerName}/resources/security"
+    into "$buildDir/resources/test/truststore/"
+    outputs.file("$buildDir/resources/test/truststore/keystore.jceks")
+}
+
+task copyDependency (type: Copy) {
+    dependsOn 'libertyCreate'
+    from configurations.mongo
+    rename { "mongo-java-driver.jar" }
+    into "$buildDir/wlp/usr/shared/resources/"
+    outputs.file("$buildDir/wlp/usr/shared/resources/mongo-java-driver.jar")
+}
+
+task startDatabase (type: Exec) {
+    group 'database'
+    description 'Execute the script to start the database.'
+
+    if (System.properties['os.name'].toLowerCase().contains('windows')) {
+        commandLine 'cmd', '/C', '../run-app/startMongo.bat', 'groupDB', groupMongoPort, 'build', mongoPath
+    } else {
+        commandLine 'sh', '../run-app/startMongo.sh', 'groupDB', groupMongoPort, 'build', mongoPath
+    }
+
+    outputs.upToDateWhen { false }
+}
+
+task stopDatabase (type: Exec) {
+    group 'database'
+    description 'Execute the script to stop the database.'
+    args = ['groupDB', mongoPath]
+
+    if (System.properties['os.name'].toLowerCase().contains('windows')) {
+        commandLine 'cmd', '/C', '../run-app/stopMongo.bat', 'groupDB', 'build', mongoPath
+    } else {
+        commandLine 'sh', '../run-app/stopMongo.sh', 'groupDB', 'build', mongoPath
+    }
+
+    outputs.upToDateWhen { false }
+}
+
+check.dependsOn 'integrationTest'
+integrationTest.dependsOn 'libertyStart', 'startDatabase', 'testKeystore'
+integrationTest.finalizedBy 'libertyStop', 'stopDatabase'
```

Comment:
is this the order they run?  Since you start liberty, then the database, when you shutdown, should you stop the database, then stop liberty?
Or would it make more sense to start the database before you start liberty?
Am trying to understand why the startup and shutdown sequences are not reflective.

Reply:
OH, do you even need startDatabase here since libertyStart also depends on startDatabase?  Maybe some of the reason for the "should" language?

Reply:
tasks listed on the same level will be executed in alphanumerical order. Some of these are simply little mistakes I overlooked